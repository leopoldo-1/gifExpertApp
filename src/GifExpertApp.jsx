
export const GiftExpertApp = () => {
  return (
    <>
      <h1>Gif Expert App</h1>
      <p>LfUzyKCYMDP4i6KcFLjVCQJmPbg4ulk0</p>
    </>
  )

}