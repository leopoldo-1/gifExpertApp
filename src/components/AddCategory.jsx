import { useState } from "react";
import PropTypes from 'prop-types'

export const AddCategory = ({ onNewCategory }) => {

  const [inputValue, setInputValue] = useState('')

  const onInputChange = ({target}) => {
    // console.log(event.target.value);
    setInputValue(target.value)
  }

  const onSubmit = (event) => {
    event.preventDefault();

    if(inputValue.trim().length <= 1) return;

    // setCategories( categories => [inputValue, ...categories])
    setInputValue('')
    onNewCategory( inputValue.trim() )
  }

  return (
    // <form onSubmit={ (event) => onSubmit(event) }>
    <form onSubmit={ onSubmit } aria-label="form">
      <input
        type="text"
        placeholder="Buscar gif"
        value={ inputValue }
        // onChange={ (event) => onInputChange(event) }
        onChange={ onInputChange }
      />
    </form>
  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}