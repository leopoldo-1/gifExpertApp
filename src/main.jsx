import React from 'react';
import ReactDOM from 'react-dom/client';
import "primereact/resources/themes/saga-blue/theme.css";  //theme
import 'primeicons/primeicons.css';

import { GiftExpertApp } from './GifExpertApp';
import './styles.css'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <GiftExpertApp />
  </React.StrictMode>
)
