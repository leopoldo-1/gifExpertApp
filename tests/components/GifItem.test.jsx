
import { render, screen } from '@testing-library/react'
import { GifItem } from '../../src/components'

describe('Tests on <GifItem/>', () => {

  const title = 'Pokemon'
  const url = 'https://www.pokemon.com/blastoise.jpg'

  test('Has to do match with the snapshot', () => {

    // Act/actuar
    const { container } = render(<GifItem title={title} url={url} />);

    // Assert/afirmar
    expect( container ).toMatchSnapshot();

  });



  test('Has to show the image with the URL and the ALT indicated', () => {

    render( <GifItem title={title} url={url} />)
    // screen.debug()

    // expect( screen.getByRole('img').alt ).toBe( title )
    const { src, alt } = screen.getByRole('img')
    expect( src ).toBe( url )
    expect( alt ).toBe( title )
  });

  test('Has to show the title on the components', () => {
    render( <GifItem title={title} url={url} />)
    expect(screen.getByText( title )).toBeTruthy();
  })


})
