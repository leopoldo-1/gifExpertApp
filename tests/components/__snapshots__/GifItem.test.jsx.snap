// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifItem/> Has to do match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Pokemon"
      src="https://www.pokemon.com/blastoise.jpg"
    />
    <p>
      Pokemon
    </p>
  </div>
</div>
`;
